Interface.C

```diff
@@ -290,8 +293,24 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh)
         // the module for volume coupling considers the mesh points in the volume and
         // on the boundary patches in order to take the boundary conditions into account
 
-        // get the number (volume centered) mesh points in the volume
-        numDataLocations_ = mesh.C().size();
+        // Get the cell labels of the overlapping region
+        std::vector<labelList> overlapCells;
+
+        if (!cellSetNames_.empty()){
+            // For every cellSet that participates in the coupling
+            for (uint j = 0; j < cellSetNames_.size(); j++)
+            {
+                // Create a cell set
+                cellSet overlapRegion(mesh, cellSetNames_[j]);
+
+                // Add the cells ID's to the vector and count how many overlap cells does the interface has
+                overlapCells.push_back(overlapRegion.toc());
+                numDataLocations_ += overlapCells[j].size();
```

Comment:
Just checking/trying to understand: What is the structure of `overlapCells`? What does `overlapRegion.toc()` return?

I understand that this part of the code tries to identify the number of cells of the mesh overlapping with each cellSet. This is needed to identify the number of coupling data locations. Correct?

Reply:
Yes, correct. `toc()` returns the keys of a `HashTable` in a list, in this case the cellIDs of the cells in the given `cellSet` (`cellSet` inherits from `HashTable`). So we add the number of cellIDs in this `cellSet` to the total number of data locations. 

I think `overlapCells` is actually unnecessary and we could directly do `numDataLocations_ += overlapRegion.toc().size()` without going through the vector.

Reply:
(`overlapCells` is actually used in the code later, so I have left it in)